File: rule-indexer-python/processors/file_processor.py
class FileProcessor:
    def __init__(self, rules):
        self.rules = rules

    def process_files(self, file_paths):
        for path in file_paths:
            try:
                with open(path, 'r') as file:
                    content = file.read()
                print(f"Processing {path}")
                for rule in self.rules:
                    rule.apply_rule(content)
            except IOError as e:
                print(f"Failed to read {path}: {e}")

File: rule-indexer-go/src/main/processors/file_processor.go
package processors

import (
	"fmt"
	"os"
	"rule-indexer-go/src/main/rules"
)

type FileProcessor struct {
    rules []rules.IndexingRule
}

func NewFileProcessor(rules []rules.IndexingRule) *FileProcessor {
    return &FileProcessor{rules: rules}
}

func (fp *FileProcessor) ProcessFiles(filePaths []string) {
    for _, path := range filePaths {
        content, err := os.ReadFile(path)
        if err != nil {
            fmt.Printf("Failed to read %s: %v\n", path, err)
            continue
        }
        fmt.Printf("Processing %s\n", path)
        for _, rule := range fp.rules {
            rule.ApplyRule(string(content))
        }
    }
}

